.changelog/24766.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+keyring: Warn if deleting a key previously used to encrypt a variable
```

Comment:
Is it worth expanding on this to note the variable should still exist? It immediately reads to me that the warning applies to any key used in the past.